airbyte-integrations/connectors/source-intercom/source_intercom/components.py

```diff
@@ -148,8 +148,10 @@ def set_initial_state(self, stream_state: StreamState):
                 self._state["prior_state"][self.parent_stream_name] = parent_stream_state
 
     def close_slice(self, stream_slice: StreamSlice, most_recent_record: Optional[Record]) -> None:
-        super().close_slice(stream_slice=stream_slice, most_recent_record=most_recent_record)
+        cursor_field = self.cursor_field.eval(self.config)
+
         if self.parent_stream:
+            self._state[cursor_field] = stream_slice[self.parent_stream_name][self.parent_cursor_field]
             self._state[self.parent_stream_name] = self.parent_stream.state
```

Comment:
nit: can you add a small comment explaining this change for future visits if needed.